modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java

```diff
@@ -137,6 +150,55 @@ protected void doProcessAction(
 			actionRequest.setAttribute(
 				ScimWebKeys.SCIM_OAUTH2_ACCESS_TOKEN, accessToken);
 		}
+		else if (Objects.equals(cmd, "reset")) {
+			OAuth2Application oAuth2Application =
+				_oAuth2ApplicationLocalService.getOAuth2Application(
+					themeDisplay.getCompanyId(), scimClientId);
+
+			_oAuth2AuthorizationService.revokeAllOAuth2Authorizations(
+				oAuth2Application.getOAuth2ApplicationId());
+
+			_oAuth2ApplicationLocalService.deleteOAuth2Application(
+				oAuth2Application);
+
+			Indexer<User> userIndexer = IndexerRegistryUtil.nullSafeGetIndexer(
+				User.class);
+
+			Indexer<UserGroup> userGroupIndexer =
+				IndexerRegistryUtil.nullSafeGetIndexer(UserGroup.class);
+
+			ActionableDynamicQuery query =
+				_expandoValueLocalService.getActionableDynamicQuery();
+
+			query.setAddCriteriaMethod(
+				dynamicQuery -> {
+					Property dataProperty = PropertyFactoryUtil.forName("data");
+
+					dynamicQuery.add(dataProperty.eq(scimClientId));
```

Comment:
Hey guys, at first glance this criteria seems weak. Correct me if I'm wrong but are you deleting an `expandoRow` just verifying that the `data` column value has the `scimClientId`? I'm concerned about deleting rows that are not related to this feature. Can we add more restrictions/conditions to the query to ensure we delete the row related to the SCIM client configuration?